diff --git a/plugins/010_envtoconf/envtoconf.sh b/plugins/010_envtoconf/envtoconf.sh
index f76a95c..7d7db30 100755
--- a/plugins/010_envtoconf/envtoconf.sh
+++ b/plugins/010_envtoconf/envtoconf.sh
@@ -2,7 +2,8 @@
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 if [ "true" == "$ENVTOCONF_ENABLED" ] || [ "$CONFIG_TYPE" == "simple" ]; then
    plugin-is-active "ENVTOCONF"
-   $DIR/envtoconf --outputdir $CONF_DIR $@
+   [[ -n $CONF_DIR ]] && mkdir -p "$CONF_DIR"
+   $DIR/envtoconf --outputdir "$CONF_DIR" $@
 fi
 
 call-next-plugin "$@"
